### PR TITLE
EE10 platform spec release review

### PR DESCRIPTION
**Pull request by starksm64**
Signed-off-by: Scott M Stark <starksm64@gmail.com>

## Specification PR template
When creating a specification project release review, create PRs with the content defined as follows.

Include the following in the PR:
- [x] A directory in the form wombat/x.y where x.y is the release major.minor version, and the directory contains the following.
- [x] Specification PDF in the form of jakarta-wombat-spec-x.y.pdf
- [x] Specification HTML in the form of jakarta-wombat-spec-x.y.html
- [x] A specification page named _index.md following the template at:
      https://github.com/jakartaee/specification-committee/blob/master/spec_page_template.md
- [x] N/A - For a Progress Review, that sufficient progress has been made on a Compatible Implementation and TCK, to ensure that the spec is implementable and testable.
- [x] For a [Release Review](https://www.eclipse.org/projects/handbook/#release-review), a summary that a Compatible Implementation is complete, passes the TCK, and that the TCK includes sufficient coverage of the specification. The TCK users guide MUST include the instructions to run the compatible implementations used to validate the release.
Instructions MAY be by reference.
   - [x] Updated release record
   - [x] Generated IP Log
   - [ ] Email to PMC
   - [x] Start release review by emailing EMO 
- [x] The URL of the OSSRH staging repository for the api, javadoc:
     https://jakarta.oss.sonatype.org/content/repositories/staging/jakarta/platform/jakarta.jakartaee-api/10.0.0/
- [x] The URL of the staging directory on downloads.eclipse.org for the proposed EFTL TCK binary:
      http://download.eclipse.org/ee4j/jakartaee-tck/jakartaee10/staged/eftl/
- [x] The URL of the compatibility certification request issue:
      https://github.com/eclipse-ee4j/jakartaee-platform/issues/514
- [x] Specification JavaDoc in the wombat/x.y/apidocs directory. 
If desired, an optional second PR can be created to contain just the JavaDoc in the `apidocs` directory.

Note: If any item does not apply, check it and mark N/A below it.

### CODE REVIEW

**Comment by ivargrimstad** [comment]
# Spec Review Checklist

1. Spec PR
  - [ ] PR uses [template](https://github.com/jakartaee/specifications/blob/master/pull_request_template.md)
  - [ ] Directory of form {spec}/x.y
  - [ ] PDF of form jakarta-{spec}-spec-x.y.pdf ("-spec" preferred but not required)
  - [ ] HTML of form jakarta-{spec}-spec-x.y.html ("-spec" preferred but not required)
  - [ ] Index page {spec}/x.y/_index.md following [template](https://github.com/jakartaee/specification-committee/blob/master/spec_page_template.md)
  - [ ] Index page {spec}/_index.md following [template](https://github.com/jakartaee/specification-committee/blob/master/spec_index_template.md)
  - [ ] No other files (e.g., no jakarta_ee_logo_schooner_color_stacked_default.png)
  - [ ] Staging repository link of the form https://jakarta.oss.sonatype.org/content/repositories/staging/jakarta/{spec}/jakarta.{spec}-api/x.y.z/
  - [ ] EFTL TCK link of the form http://download.eclipse.org/.../+.zip
  - [ ] Compatibility certification link of the form https://github.com/eclipse-ee4j/{project}/#{issue}
  - [ ] (Optional) Second PR for just apidocs

2. _index.md
  - [ ] Link to project release plan of the form https://projects.eclipse.org/projects/ee4j.{spec}/releases/x.y/plan
  - [ ] Link to spec pdf
  - [ ] Link to spec html
  - [ ] Link to apidocs
  - [ ] Link to final TCK download zip file of the form https://download.eclipse.org/jakartaee/{spec}/x.y/*{spec}-tck-x.y.z.zip (The folder path is required, the file name pattern is preferred.)
  - [ ] Link to API jar file of the form https://search.maven.org/artifact/jakarta.{spec}/jakarta.{spec}-api/x.y.z/jar
  - [ ] Name of and link to at least one Compatible Implementation
  - [ ] Review summary statement to ensure it has been updated to reflect the work accomplished and does not contain proposal statements as might be found in Plan Review statement.

3. javadocs
  - [ ] Footer contains Eclipse copyright and link to license
  - [ ] ESFL license is included, usually as doc-files/speclicense.html
  - [ ] no META-INF directory in PR
  - [ ] javadocs-jar artifact matches apidocs (optional for this release)

4. Spec PDF
  - [ ] Correct spec title
  - [ ] Version number of the form x.y, not x.y.z
  - [ ] Correct Eclipse copyright line
  - [ ] No DRAFT or SNAPSHOT
  - [ ] Correct Logo

5. Spec HTML
  - [ ] Same as PDF

6. TCK zip file
  - [ ] README file (optional for this release)
  - [ ] EFTL license file, preferably named LICENSE.md
  - [ ] User's Guide (or equivalent documentation)
  - [ ] How to test the Compatible Implementation(s) listed in _index.md above with the TCK (may be in UG)

7. TCK User's Guide (or equivalent documentation)
  - [ ] Software requirements listed
  - [ ] Installation and configuration described
  - [ ] How to run tests
  - [ ] Where to file challenges

8. Compatibility certification request
  - [ ] Request follows [template](https://github.com/jakartaee/specification-committee/blob/master/compatibility-certification-request.md)
  - [ ] SHA-256 fingerprint matches staged TCK zip file
  - [ ] Request issue has `certification` label.

9. TCK results summary
  - [ ] Page is hosted by Compatible Implementation project
  - [ ] Includes all information from certification request
  - [ ] Summary includes number of tests passed, failed, errors
  - [ ] SHA-256 fingerprint matches staged TCK zip file on cert request

10. If a Release Review is required, the specification project team contacts the EMO to initiate the review by sending an email to emo@eclipse.org.
    (A Release Review is not required if the current release is a Service Release based on a previously successful Major or Minor
    release as indicated by a release record on the project's Releases page, e.g., the [Jakarta Servlet releases page](https://projects.eclipse.org/projects/ee4j.servlet/reviews).)
    - [ ] An issue will be created by the EMO to track the release review.
    - [ ] The specification project team requests approval for the release from the PMC by sending an email to ee4j-pmc@eclipse.org.
    - [ ] The specification project team then delivers an IP Log to the IP Team for their review as described in https://www.eclipse.org/projects/handbook/#pmi-commands-iplog.

11. Update Jakarta EE API jar
  - [ ] Update the Jakarta EE API jar by submitting a PR to the jakartaee-api project that updates the version number of your API jar file.

**Comment by OndroMih** [comment]
CCR for GF is here: https://github.com/eclipse-ee4j/jakartaee-platform/issues/514

@starksm64 , @ivargrimstad , can you add it to the description of this PR?

With this, I believe we can start preparing the ballot with the last remaining points:

* Updated release record
* Generated IP Log
* Email to PMC
* Start release review by emailing EMO

**Comment by ivargrimstad** [comment]
The Spec Documents (PDF and HTML) have the status "DRAFT". Must be regenerated to set the status to "Final"

**Comment by OndroMih** [comment]
Hi @starksm64, can you resolve conflicts? Can we then move this PR out of draft mode? Or is there still something to do before asking for a ballot?

**Comment by starksm64** [comment]
It has been marked ready for review. I'll go over the full checklist tomorrow or this weekend.

**Comment by starksm64** [comment]
IP log: http://dev.eclipse.org/ipzilla/show_bug.cgi?id=24185

**Comment by ivargrimstad** [comment]
The only things missing now, as far as I can tell are:

- [ ] a link to GlassFish 7.0.0-M7 as the compatible implementation used for ratification
- [ ] Update the SHA for the TCK run on the [TCK Results page](https://glassfish.org/certifications/jakarta-platform/10/TCK-Results-7.0) to `62a9cbbf46315ef4b5487fdbdf6e628d272fa3ace9148c5716ba57ea4415df19`

**Comment by arjantijms** [comment]
The link to GlassFish 7.0.0-M7 is https://download.eclipse.org/ee4j/glassfish/glassfish-7.0.0-M7.zip

**Comment by OndroMih** [comment]
The TCK Results page is now updated: https://glassfish.org/certifications/jakarta-platform/10/TCK-Results-7.0.0-M7.

**Comment by ivargrimstad** [comment]
Everything looks fine from my side now, so I will start the ballot on Monday if nothing else comes up.

File: platform/10/_index.md [hunk 1]
```diff
@@ -1,28 +1,29 @@
...
-* [Jakarta EE Platform 10 Specification Document](./jakarta-platform-spec-10.html) (HTML)
+* [Jakarta EE Platform 10 Specification Document](./jakarta-platform-spec-10.0.pdf) (PDF)
+* [Jakarta EE Platform 10 Specification Document](./jakarta-platform-spec-10.0.html) (HTML)
 * [Jakarta EE Platform 10 Javadoc](./apidocs)
 * [Jakarta EE Platform 10 TCK](https://download.eclipse.org/jakartaee/platform/10/jakarta-jakartaeetck-10.0.0.zip) ([sig](https://download.eclipse.org/jakartaee/platform/10/jakarta-jakartaeetck-10.0.0.zip.sig),[sha](https://download.eclipse.org/jakartaee/platform/10/jakarta-jakartaeetck-10.0.0.zip.sha256),[pub](https://jakarta.ee/specifications/jakartaee-spec-committee.pub))
 * Maven coordinates
   * [jakarta.platform:jakarta.jakartaee-api:jar:10.0.0](https://search.maven.org/artifact/jakarta.platform/jakarta.jakartaee-api/10.0.0/jar)
 * Compatible Implementations used for [ratification](https://www.eclipse.org/projects/efsp/?version=1.2#efsp-ratification).
-  * [TBD]()
+  * [GlassFish-7.0.0-M7](https://download.eclipse.org/ee4j/glassfish/glassfish-7.0.0-M7.zip)
```
Comment (ivargrimstad) [inline platform/10/_index.md:20]: Update to M8

Comment (edbratt) [reply platform/10/_index.md:20]: Link to M8: https://download.eclipse.org/ee4j/glassfish/glassfish-7.0.0-M8.zip

Comment (starksm64) [reply platform/10/_index.md]: This should be ready

**Review by ivargrimstad — APPROVED** [review]

**Comment by ivargrimstad** [comment]
# Mentor Checklist 
 - [ ] adds this final checklist to the main PR.
 - [ ] adds the `approved` label to the PRs, and sends out the Ballot Summary per this [template](#ballot-summary-email-template) to the [public Jakarta EE Specification Committee email list](jakarta.ee-spec@eclipse.org)
 - [ ] calculates the staged EFTL TCK signature and promotes it to the committee download area
  using the https://ci.eclipse.org/jakartaee-spec-committee/job/promote-release/ job. Manually editing the jenkins Build Information will help identify the build (ie. Mail 2.0 or CDI 3.0).
 - [ ] merges the specification (and apidocs) PRs, ensuring the "date:" field in the _index.md file has an appropriate value to allow publishing.
 - [ ] updates the specification page with the ballot results. This is normally done via a separate PR that should be reviewed, approved, and merged.
 - [ ] notifies the EMO of the ballot results by email to [emo@eclipse-foundation.org](emo@eclipse-foundation.org). Just forward the ballot summary note sent earlier to the public Spec Committee email list.
 - [ ] creates an issue in the specification project that includes the following checklist for the specification project team:
https://github.com/eclipse-ee4j/jakartaee-platform/issues/542

**Comment by ivargrimstad** [comment]
@starksm64 Do you think you can resolve the merge conflicts and merge this PR?
Not sure if I have access to do commits to the branch

**Comment by starksm64** [comment]
Resolved

**Pull request merged**